File: backend/ms_auth/src/main/java/wevioo/tn/ms_auth/entities/Team.java
package wevioo.tn.ms_auth.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Data
@Entity
@Table(name="Teams")
@NoArgsConstructor
public class Team {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    private String avatar;
    @ManyToMany(fetch = FetchType.LAZY, cascade = { CascadeType.PERSIST, CascadeType.MERGE })
    @JoinTable(
            name = "Team_Member",
            joinColumns = @JoinColumn(name = "team_id"),
            inverseJoinColumns = @JoinColumn(name = "member_id")
    )
    private Set<Member> members = new HashSet<>();
    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private UserEntity userTeam;
}

File: frontend/src/models/user/Team.ts
import { IAddContact } from "./Contact"

export interface Team {
    id?:number,
    name: string,
    description:string,
    avatar:string|null,
}

export interface ITeam {
    id?:number,
    name: string,
    description:string,
    avatar:string|null|undefined,
}

export interface IUpdateTeam extends ITeam{
    members:number[]
} 

export interface ITeamWithContact extends ITeam{
    members:IAddContact[]
} 